deps/rabbit/src/rabbit_amqqueue.erl

```diff
@@ -1578,11 +1580,13 @@ pid_of(Q) -> amqqueue:get_pid(Q).
 -spec pid_of(rabbit_types:vhost(), rabbit_misc:resource_name()) ->
           pid() | rabbit_types:error('not_found').
 
-pid_of(VHost, QueueName) ->
-  case lookup(rabbit_misc:r(VHost, queue, QueueName)) of
+pid_of(VHost, QueueName) when is_binary(VHost) ->
+  case lookup(VHost, QueueName) of
     {ok, Q}                -> pid_of(Q);
     {error, not_found} = E -> E
-  end.
+  end;
+pid_of(default_vhost, QueueName) ->
```

Comment:
This clause does not make much sense for an Erlang shell user. Please turn it into a `pid_of/1`.

Reply:
I have done it; thanks for your feedback.

Reply:
My comment was not actually addressed.

`pid_of/2` with an artificial special case first argument does not make much sense.

``` erl
pid_of(QueueName) when is_binary(QueueName) ->
    VHost = rabbit_vhost:default_name(),
    case lookup(VHost, QueueName) of
    	{ok, Q}                -> pid_of(Q);
	    {error, not_found} = E -> E
    end.
```

If the virtual host is not explicitly given, and we only have a queue name, try looking it up in the default virtual host. If we are given a record, then it returns the pid value stored there.

Not very elegant but will break nothing and does not require us coming up with a clever short name for this new alternative to `pid_of/1` which accepts a record or `pid_of/2` which accepts a virtual host name.

Reply:
Thanks for this explanation, @michaelklishin; I did understand what you meant.